src/rust/bin/wallet_tool.rs

```diff
@@ -0,0 +1,580 @@
+                let line = prompt(&mut stdin, &mut buf)?;
+                if words[n] == line {
+                    break;
+                }
+                println!("That's not correct, please try again.");
+            }
+        }
+        Ok(())
+    })();
+
+    panic::set_hook(old_hook);
+    clear_and_show_cautions(export_path);
+    res?;
+
+    let mut cli_args = cli_options;
+    cli_args.extend_from_slice(&["walletconfirmbackup".to_string(), phrase.to_string()]);
+
+    // Always pass false for debug to avoid printing the recovery phrase.
+    exec(&zcash_cli, &cli_args, false)
+        .and_then(|out| {
+            let cli_err: Vec<_> = from_utf8(&out.stderr)
+                .with_context(|| "Output from zcash-cli was not UTF-8")?
+                .lines()
+                .map(|s| s.trim_end_matches('\r'))
+                .collect();
+            if opts.debug {
+                eprintln!("DEBUG: stderr {:?}", cli_err);
+            }
+            if !cli_err.is_empty() {
+                if cli_err[0].starts_with("error: couldn't connect") {
+                    println!("\nWe could not connect to zcashd; it may have exited.");
+                    return Err(WalletToolError::ZcashdConnection.into());
+                } else {
+                    println!(
+                        "\nThere was an unexpected response from zcashd:\n> {}",
+                        cli_err.join("\n> "),
+                    );
+                    return Err(WalletToolError::UnexpectedResponse.into());
+                }
+            }
+            println!(concat!(
+                "\nThe backup of the emergency recovery phrase for the zcashd\n",
+                "wallet has been successfully confirmed 🙂. You can now use the\n",
+                "zcashd RPC methods that create keys and addresses in that wallet.\n\n",
+                "If you use other wallets, their recovery information will need\n",
+                "to be backed up separately.\n"
+            ));
+            Ok(())
+        })
+        .map_err(|e| {
+            println!(concat!(
+                "\nzcash-wallet-tool was unable to communicate to zcashd that the\n",
+                "backup was confirmed. This can happen if zcashd stopped, in which\n",
+                "case you should try again. If zcashd is still running, please seek\n",
+                "help or try to use 'zcash-cli -stdin walletconfirmbackup' manually.\n"
+            ));
+            e
+        })?;
+    Ok(())
+}
+
+fn prompt<'a>(input: &mut Stdin, buf: &'a mut String) -> anyhow::Result<&'a str> {
+    input
+        .read_line(buf)
+        .with_context(|| "Error reading from stdin")?;
+
+    if !buf.ends_with('\n') {
+        Err(io::Error::from(ErrorKind::UnexpectedEof))
+            .with_context(|| "End of file reading from stdin")
+    } else {
+        Ok(buf.trim_end_matches(|c| c == '\r' || c == '\n').trim())
+    }
+}
```

Comment:
So the first half of this function becomes something like:
```suggestion
pub fn prompt(input: &mut Stdin) -> anyhow::Result<SecretString> {
    let mut buf = String::with_capacity(MAX_USER_INPUT_LEN);
    let res = input.read_line(&mut buf)
        .with_context(|| "Error reading from stdin");
    // Ensure the buffer is zeroized even on error.
    let line = SecretString::new(buf);
    res.map(|_| line)
}
```